### ISSUE TITLE
How to download model

### ISSUE THREAD

**Issue by adnosi3000** [davidgeorgewilliams/Aerial-YOLO-DOTA#2]
Hi, I wolud like tu use the model but I nned it in .onnx format. Can you please suggest any pipeline to convert it? Which file should I download and convert?

**Comment by davidgeorgewilliams** [comment]
Hi @adnosi3000,

Thanks for your interest in our Aerial-YOLO-DOTA project! I'd be happy to help you convert the model to ONNX format.

## Converting the YOLOv9 Model to ONNX

The process is quite straightforward using the Ultralytics API. Here's how you can do it:

### Step 1: Download the trained model

You can find our best performing model in the `results/100e-16b-0.01lr` directory of the repository. Download the `best.pt` file from there.

### Step 2: Install required dependencies

Make sure you have the necessary packages installed:

```bash
pip install ultralytics onnx onnxruntime
```

### Step 3: Convert to ONNX format

Create a simple Python script with the following code:

```python
from ultralytics import YOLO

# Load the trained model
model = YOLO('path/to/best.pt')  # Replace with your downloaded model path

# Export to ONNX
model.export(format='onnx', simplify=True)
```

This will generate a `best.onnx` file in the same directory as your `.pt` file.

### Advanced Options (if needed)

If you need more control over the export process, you can use additional parameters:

```python
# Export with specific parameters
model.export(
    format='onnx',           # Export to ONNX format
    imgsz=640,               # Image size (match what we used in training)
    half=True,               # Use FP16 for better performance
    simplify=True,           # Simplify the model structure
    opset=12,                # ONNX opset version
    dynamic=True,            # Enable dynamic axes for variable batch sizes
    workspace=4              # Workspace size in GB
)
```

### Step 4: Verify the ONNX model (recommended)

It's always a good idea to verify that the exported model is valid:

```python
import onnx

# Load and check the ONNX model
onnx_model = onnx.load('path/to/best.onnx')
onnx.checker.check_model(onnx_model)
print("ONNX model is valid!")
```

## Using the ONNX Model

Once converted, you can use the ONNX model with various inference libraries such as ONNX Runtime or OpenCV's DNN module. The model expects input images with size 640x640 and normalized pixel values (0-1).

Let me know if you run into any issues during the conversion process, and I'd be happy to help troubleshoot!

Best regards,
David

**Comment by Nuzhny007** [comment]
Hi!
Thnx for this interesting project!

> You can find our best performing model in the `results/100e-16b-0.01lr` directory of the repository. Download the `best.pt` file from there.

But here has not any best.pt: https://github.com/davidgeorgewilliams/Aerial-YOLO-DOTA/tree/main/results/100e-16b-0.01lr
Do you can share own model?

**Comment by AjayKumarGogineni777** [comment]
Hey, Thank you for sharing the training pipeline and code. I do not have enough resources to train the model. It would be really helpful if you can share the weights of the trained model. I cannot find the best.pt file in the results/100e-16b-0.01lr directory. Thank you!

**Comment by degD** [comment]
Hello, I came across this project while researching aerial vehicle detection, are there any plans to release the pretrained models?

**Comment by AjayKumarGogineni777** [comment]
I have trained the model for 250 epochs and uploaded the weights here: https://drive.google.com/drive/folders/1gf9kIjkFEy14AoHulqeBJQsxOE2PgF9j?usp=drive_link
